packages/application-to-register-webapp/src/routes/__tests__/land/add-legal-agreement-parties.spec.js

```diff
@@ -0,0 +1,134 @@
+      const request = {
+        yar: redisMap,
+        info: {
+          referrer: 'check-legal-agreement-details'
+        }
+      }
+      const legalAgreementParties = require('../../land/add-legal-agreement-parties')
+      await legalAgreementParties.default[0].handler(request, h)
+
+      postOptions.payload = {
+        'organisation[0][organisationName]': 'Bambury',
+        'organisation[0][role]': 'County Council',
+        otherPartyName: ''
+      }
+      const response = await submitPostRequest(postOptions)
+      expect(response.statusCode).toBe(302)
+    })
+
+    it('should add multiple legal party to legal agreement', async () => {
+      postOptions.payload = {
+        'organisation[0][organisationName]': 'Sun',
+        'organisation[0][role]': 'Developer',
+        'organisation[1][organisationName]': 'Intel',
+        'organisation[1][role]': 'County Council',
+        'organisation[2][organisationName]': 'Intelij',
+        'organisation[2][role]': 'Landowner',
+        'organisation[3][organisationName]': 'Intelijet',
+        'organisation[3][role]': 'Responsible body',
+        otherPartyName: [
+          '',
+          ''
+        ]
+      }
+      const response = await submitPostRequest(postOptions)
+      expect(response.statusCode).toBe(302)
+    })
+
+    it('should add multiple legal organisation with other party choice to legal agreement', async () => {
+      postOptions.payload = {
+        'organisation[0][organisationName]': 'Test',
+        'organisation[0][role]': 'Other',
+        otherPartyName: [
+          'party One',
+          'Party two'
+        ],
+        'organisation[1][organisationName]': 'Test Two',
+        'organisation[1][role]': 'Other'
+      }
+      const response = await submitPostRequest(postOptions)
+      expect(response.statusCode).toBe(302)
+    })
+
+    it('should fail to add single legal party to legal agreement without organisation name', async () => {
+      postOptions.payload = {
+        'organisation[0][organisationName]': '',
+        'organisation[0][role]': 'Developer',
+        otherPartyName: ''
+      }
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add single legal party to legal agreement without organisation name', async () => {
+      postOptions.payload = {
+        'organisation[0][organisationName]': '',
+        'organisation[0][role]': 'Developer',
+        otherPartyName: ''
+      }
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add single legal party to legal agreement without organisation name nor role', async () => {
+      postOptions.payload = {
+        'organisation[0][organisationName]': '',
+        otherPartyName: ''
+      }
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
```

Comment:
Please can you add additional checks for expected errors. [For example](https://github.com/DEFRA/biodiversity-net-gain-service/blob/master/packages/application-to-register-webapp/src/routes/__tests__/land/add-landowners.spec.js).